File: logitrack_core/internal/model/access_log.go
package model

import "time"

type AccessEventType string

const (
	AccessEventLoginSuccess           AccessEventType = "login_success"
	AccessEventLoginFailure           AccessEventType = "login_failure"
	AccessEventLogout                 AccessEventType = "logout"
	AccessEventPasswordResetRequested AccessEventType = "password_reset_requested"
	AccessEventPasswordResetConfirmed AccessEventType = "password_reset_confirmed"
)

type AccessLog struct {
	ID            string          `json:"id"`
	Username      string          `json:"username"`
	UserID        string          `json:"user_id,omitempty"`
	Role          string          `json:"role,omitempty"`
	EventType     AccessEventType `json:"event_type"`
	IPAddress     string          `json:"ip_address,omitempty"`
	Country       string          `json:"country,omitempty"`
	City          string          `json:"city,omitempty"`
	Result        string          `json:"result,omitempty"`
	FailureReason string          `json:"failure_reason,omitempty"`
	Timestamp     time.Time       `json:"timestamp"`
}

type AccessLogFilter struct {
	Username string
	DateFrom string
	DateTo   string
	Limit    int
}

File: logitrack_web/src/api/accessLog.ts
import { api } from "./auth";

export type AccessEventType =
  | "login_success"
  | "login_failure"
  | "logout"
  | "2fa_required"
  | "password_reset_requested"
  | "password_reset_confirmed";

export interface AccessLog {
  id: string;
  username: string;
  user_id: string;
  role: string;
  event_type: AccessEventType;
  ip_address: string;
  country: string;
  city: string;
  result: string;
  failure_reason?: string;
  timestamp: string;
}

export interface AccessLogFilters {
  username?: string;
  date_from?: string;
  date_to?: string;
}

export const accessLogApi = {
  list: (limit: number, filters?: AccessLogFilters) =>
    api
      .get<AccessLog[]>("/admin/access-logs", {
        params: { limit, ...filters },
      })
      .then((r) => r.data),
};